micro_sam/evaluation/livecell.py

```diff
@@ -238,137 +316,103 @@ def run_livecell_instance_segmentation_with_decoder(
-    parser.add_argument("-c", "--ckpt", type=str, required=True,
-                        help="Provide model checkpoints (vanilla / finetuned).")
-    parser.add_argument("-i", "--input", type=str, required=True,
-                        help="Provide the data directory for LIVECell Dataset.")
-    parser.add_argument("-e", "--experiment_folder", type=str, required=True,
-                        help="Provide the path where all data for the inference run will be stored.")
-    parser.add_argument("-m", "--model", type=str, required=True,
-                        help="Pass the checkpoint-specific model name being used for inference.")
+    parser.add_argument(
+        "-c", "--ckpt", type=str, required=True,
+        help="Provide model checkpoints (vanilla / finetuned)."
+    )
+    parser.add_argument(
+        "-i", "--input", type=str, required=True,
+        help="Provide the data directory for LIVECell Dataset."
+    )
+    parser.add_argument(
+        "-e", "--experiment_folder", type=str, required=True,
+        help="Provide the path where all data for the inference run will be stored."
+    )
+    parser.add_argument(
+        "-m", "--model", type=str, required=True,
+        help="Pass the checkpoint-specific model name being used for inference."
+    )
 
     # the experiment type:
-    # - default settings (p1-n0, p2-n4, p4-n8, box)
-    # - full experiment (ranges: p:1-16, n:0-16)
-    # - automatic mask generation (auto)
-    # if none of the two are active then the prompt setting arguments will be parsed
-    # and used to run inference for a single prompt setting
-    parser.add_argument("-d", "--default_experiment", action="store_true")
-    parser.add_argument("-f", "--full_experiment", action="store_true")
-    parser.add_argument("-a", "--auto_mask_generation", action="store_true")
-
-    # the prompt settings for an individual inference run
-    parser.add_argument("--box", action="store_true", help="Activate box-prompted based inference")
-    parser.add_argument("--points", action="store_true", help="Activate point-prompt based inference")
-    parser.add_argument("-p", "--positive", type=int, default=1, help="No. of positive prompts")
-    parser.add_argument("-n", "--negative", type=int, default=0, help="No. of negative prompts")
-
-    # optional external prompt folder
-    parser.add_argument("--prompt_folder", help="Provide the path where all input point prompts will be stored")
+    # 1. precompute image embeddings
+    # 2. interactive instance segmentation (interactive)
+    #     - iterative prompting
+    #         - starting with point
+    #         - starting with box
+    # 3. automatic segmentation (auto)
+    #     - automatic mask generation (amg)
+    #     - automatic instance segmentation (ais)
+
+    parser.add_argument("-p", "--precompute_embeddings", action="store_true")
+    parser.add_argument("-ip", "--interactive_segmentation", action="store_true")
```

Comment:
@constantinpape Just to be sure, I call the iterative prompt-based inference as interactive segmentation (to keep the name simple and to-the-point). Does this work, or should I update this to iterative prompting as well?

Reply:
Fixed it. Thanks!